### PR TITLE
CIRC-957- Declared lost, Anonymized, Renewed through override loan actions don't appear in circ log

### PR DESCRIPTION
**Pull request by siarhei-charniak**
[CIRC-957](https://issues.folio.org/browse/CIRC-957) - Declared lost, Anonymized, Renewed through override loan actions don't appear in circ log

## Purpose

Declared lost, Anonymized, Renewed through override loan actions should appear in circ log

## Approach
* fixed anonymize loan log event

## Pre-Merge Checklist:
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code is 3% or less
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [ ] There are no breaking changes in this PR.
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.  

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/main/java/org/folio/circulation/domain/representations/logs/LoanLogContext.java [hunk 1]
```diff
@@ -57,6 +58,15 @@ public static LoanLogContext from(Loan loan) {
     return loanLogContext;
   }
 
+  public static LoanLogContext fromAnonymize(Loan loan) {
+    return new LoanLogContext()
+      .withItemId(getProperty(loan.asJson(), ITEM_ID.value()))
+      .withAction("Anonymize")
+      .withDate(DateTime.now())
```
Comment (bohdan-suprun) [inline src/main/java/org/folio/circulation/domain/representations/logs/LoanLogContext.java:65]:
```suggestion
      .withDate(DateTime.now(UTC))
```

File: src/main/java/org/folio/circulation/domain/representations/logs/LoanLogContext.java [hunk 2]
```diff
@@ -57,6 +58,15 @@ public static LoanLogContext from(Loan loan) {
...
+      .withLoanId(getProperty(loan.asJson(), "id"))
```
Comment (bohdan-suprun) [inline src/main/java/org/folio/circulation/domain/representations/logs/LoanLogContext.java:66]:
```suggestion
      .withLoanId(loan.getId())
```

File: src/main/java/org/folio/circulation/domain/representations/logs/LoanLogContext.java [hunk 3]
```diff
@@ -57,6 +58,15 @@ public static LoanLogContext from(Loan loan) {
...
+      .withServicePointId(getProperty(loan.asJson(), "checkinServicePointId"));
```
Comment (bohdan-suprun) [inline src/main/java/org/folio/circulation/domain/representations/logs/LoanLogContext.java:67]:
```suggestion
      .withServicePointId(loan.asJson().getCheckInServicePointId());
```

File: src/main/java/org/folio/circulation/services/EventPublisher.java [hunk 4]
```diff
@@ -218,7 +218,7 @@ public EventPublisher(PubSubPublishingService pubSubPublishingService) {
   }
 
   public CompletableFuture<Result<Void>> publishAnonymizeEvent(Loan loan) {
-    return publishLogRecord(LoanLogContext.from(loan).withAction("Anonymize").withDescription("").asJson(), LOAN);
+    return publishLogRecord(LoanLogContext.fromAnonymize(loan).asJson(), LOAN);
```
Comment (bohdan-suprun) [inline src/main/java/org/folio/circulation/services/EventPublisher.java:221]: Is there a test for this change?

Comment (siarhei-charniak) [reply src/main/java/org/folio/circulation/services/EventPublisher.java:221]: This change does not require test

**Review by khandramai — APPROVED** [review]

**Pull request merged**